system/storage/copy.go

```diff
@@ -34,11 +34,12 @@ func (s *service) Copy(context *endly.Context, request *CopyRequest) (*CopyRespo
 func (s *service) copy(context *endly.Context, request *CopyRequest, response *CopyResponse) error {
 	var udfModifier option.Modifier
 	if request.Udf != "" {
+		var ok bool
 		UDF, err := udf.TransformWithUDF(context, request.Udf, "", nil)
 		if err != nil {
-			return errors.Wrapf(err, "failed to Copy")
+			return errors.Wrapf(err, "failed to get udf: %v", request.Udf)
```

Comment:
why not consistency use either fmt.Errorf or errors.New() or errors.Wrap?

Reply:
Wrapf preserves error stack

Reply:
https://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully